### PR TITLE
updateGuildPreferences call informs whether available songs may change [#376]

### PR DESCRIPTION
**Pull request by taahamahdi**
is this better or worse than using true/false :thinking:

### CODE REVIEW

File: src/models/guild_preference.ts [hunk 1]
```diff
@@ -38,6 +38,11 @@ interface GameOptions {
     guessTimeout: number;
 }
 
+const enum AvailableSongs {
```
Comment (Brainicism) [inline src/models/guild_preference.ts:41]: Use boolean

File: src/models/guild_preference.ts [hunk 2]
```diff
@@ -223,11 +231,13 @@ export default class GuildPreference {
         return this.gameOptions.shuffleType === ShuffleType.UNIQUE;
     }
 
-    async updateGuildPreferences(_db: Knex) {
+    async updateGuildPreferences(_db: Knex, songsState: AvailableSongs) {
```
Comment (Brainicism) [inline src/models/guild_preference.ts:234]: songsState => `songListModified`

File: src/models/guild_preference.ts [hunk 3]
```diff
@@ -223,11 +231,13 @@ export default class GuildPreference {
...
         await _db("guild_preferences")
             .where({ guild_id: this.guildID })
             .update({ guild_preference: JSON.stringify(this) });
         const gameSession = state.gameSessions[this.guildID];
-        if (gameSession) gameSession.resetLastPlayedSongsQueue();
+        if (gameSession && songsState === AvailableSongs.Modifies) {
+            gameSession.resetLastPlayedSongsQueue();
```
Comment (Brainicism) [inline src/models/guild_preference.ts:240]: Don't we still have this logic in gamesession

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/models/guild_preference.ts (hunk 3 above)
Comment (taahamahdi) [reply src/models/guild_preference.ts:240]: That call to reset is when the user plays all of their unique songs, so it's still needed

File: src/models/guild_preference.ts [hunk 4]
```diff
@@ -67,17 +67,20 @@ export default class GuildPreference {
         }
         if (gameOptionModified) {
             this.updateGuildPreferences(dbContext.kmq);
+            this.updateGameSession(false);
```
Comment (Brainicism) [inline src/models/guild_preference.ts:70]: Game session doesn't exist at this point, not needed here

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/models/guild_preference.ts (hunk 4 above)
Comment (taahamahdi) [reply src/models/guild_preference.ts:70]: Thanks!

**Review by Brainicism — APPROVED** [review]

**Pull request merged**